packages/zambdas/src/subscriptions/questionnaire-response/ai-interview-summary/index.ts

```diff
@@ -0,0 +1,170 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+import { Condition, Encounter, Observation, Questionnaire, QuestionnaireResponse } from 'fhir/r4b';
+import { AI_OBSERVATION_META_SYSTEM, createOystehrClient, getSecret, Secrets, SecretsKeys } from 'utils';
+import {
+  configSentry,
+  getAuth0Token,
+  makeDiagnosisConditionResource,
+  makeObservationResource,
+  parseCreatedResourcesBundle,
+  saveResourceRequest,
+  validateJsonBody,
+  ZambdaInput,
+} from '../../../shared';
+import Oystehr, { BatchInputPostRequest } from '@oystehr/sdk';
+import { invokeChatbot } from '../../../shared/ai';
+
+export const INTERVIEW_COMPLETED = 'Interview completed.';
+
+const PROMPT = `I'll give you a transcript of a chat between a healthcare provider and a patient. 
+Please generate history of present illness, past medical history, past surgical history, medications history, 
+allergies, social history, family history, hospitalization history and potential diagnoses with ICD-10 codes for the patient. 
+Please present a response in JSON format. Don't add markdown. Use property names in camel case. For ICD-10 codes use "icd10" property.  
+Use a single string property in JSON for each section except potential diagnoses. 
+The transcript: `;
+
+const AI_RESPONSE_KEYS = [
+  'historyOfPresentIllness',
+  'pastMedicalHistory',
+  'pastSurgicalHistory',
+  'medicationsHistory',
+  'allergies',
+  'socialHistory',
+  'familyHistory',
+  'hospitalizationHistory',
+];
+
+let oystehrToken: string;
+
+interface Input {
+  questionnaireResponse: QuestionnaireResponse;
+  secrets: Secrets | null;
+}
+
+export const index = async (input: ZambdaInput): Promise<APIGatewayProxyResult> => {
+  configSentry('sub-ai-interview-summary', input.secrets);
+  console.log('AI interview summary invoked');
+  console.log(`Input: ${JSON.stringify(input)}`);
+  try {
+    const { questionnaireResponse, secrets } = validateInput(input);
+    const chatTranscript = createChatTranscript(questionnaireResponse);
+    const aiResponseString = (
+      await invokeChatbot([{ role: 'user', content: PROMPT + '\n' + chatTranscript }], secrets)
+    ).content.toString();
+    console.log(`AI response: "${aiResponseString}"`);
+    const aiResponse = JSON.parse(aiResponseString);
+    const oystehr = await createOystehr(secrets);
+    const encounter = await oystehr.fhir.get<Encounter>({
+      resourceType: 'Encounter',
+      id: questionnaireResponse.encounter?.reference?.split('/')[1] ?? '',
+    });
+    const requests: BatchInputPostRequest<Observation | Condition>[] = [];
+    const patientId = encounter.subject?.reference?.split('/')[1];
+    requests.push(...createObservations(aiResponse, encounter.id!, patientId!));
```

Comment:
I think it's better to check all required data and made a return with logging errors early and then avoid using `! syntax` because type guarding will work.